contract/src/main.rs

```diff
@@ -1560,57 +1555,61 @@ pub extern "C" fn set_token_metadata() {
 
 #[no_mangle]
 pub extern "C" fn migrate() {
-    let requires_rlo_migration = utils::requires_rlo_migration();
+    let reporting_mode = if runtime::get_key(REPORTING_MODE).is_some() {
+        Some(utils::get_reporting_mode())
+    } else {
+        None
+    };
 
-    if !requires_rlo_migration && runtime::get_key(RLO_MFLAG).is_some() {
-        runtime::revert(NFTCoreError::ContractAlreadyMigrated)
-    }
+    if [None, Some(OwnerReverseLookupMode::NoLookUp)].contains(&reporting_mode) {
```

Comment:
won't this condition always be met cause of the preceding lines

Reply:
Do you prefer maybe following ? it may be more explicit but then it depends on future possible OwnerReverseLookupMode if any.

```
if Some(OwnerReverseLookupMode::Complete) != reporting_mode {
```

Reply:
I'd suggest trying to have the logic determine the early exit case, if you can't find the reporting mode, then we must perform the migration, if there reporting mode is available, then just alter the total token supply if the mode is no look up and exit

Reply:
It's not really exit cases but I divided the migration into tow functions do_migration() and update_token_supply(), with do_migration calling update_token_supply. Let me know if this version reads better
https://github.com/casper-ecosystem/cep-78-enhanced-nft/blob/c9f11a7776adb63e4b44f9e339e0e1faee072a7a/contract/src/main.rs#L1661-L1678

Reply:
I am merging this last version for now before code freeze, let me know if some changes still needed.